### ISSUE TITLE
Would you like manpages?

### ISSUE THREAD

**Issue by baldurmen** [spl0k/supysonic#155]
Heya,

I wrote manpages for the supysonic Debian package (efforts ongoing!) and I wanted to know if you wanted me to push them upstream somewhere.

You really don't have to say yes if you don't find that useful - they'll be part of the Debian package anyway. AFAIK you can't push manpages cleanly using setup.py, but I thought I might ask.

I'm using `rst2man` (part of `docutils`) to generate manpages from RST files. Apparently Github hates attachments that aren't pictures, so here's what they look like. You can save them locally and render them this way:

    man .\supysonic-cli.1

`supysonic-cli.1`
```
.\" Man page generated from reStructuredText.
.
.TH SUPYSONIC-CLI 1 "2019" "" ""
.SH NAME
supysonic-cli \- a Python implementation of the Subsonic server API
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.nf
supysonic\-cli [\fBsubcommand\fP]
supysonic\-cli \fBhelp\fP
supysonic\-cli \fBhelp\fP \fIuser\fP
supysonic\-cli \fBhelp\fP \fIfolder\fP
.fi
.sp
.SH DESCRIPTION
.nf
supysonic is a Python implementation of the Subsonic server API.
Current supported features are:
.fi
.sp
.nf

.fi
.sp
.nf
* browsing (by folders or tags)
* streaming of various audio file formats
* transcoding
* user or random playlists
* cover arts (as image files in the same folder as music files)
* starred tracks/albums and ratings
* Last.FM scrobbling
.fi
.sp
.nf
The "Subsonic API" is a set of adhoc standards to browse, stream or
download a music collection over HTTP.
.fi
.sp
.SH SUBCOMMANDS
.nf
If ran without arguments, \fBsupysonic\-cli\fP will open an interactive
prompt.
.fi
.sp
.sp
\fBsupysonic\-cli\fP has three different subcommands:
.nf

.fi
.sp
.nf
* help
* user
* folder
.fi
.sp
.nf
For more details on the \fBuser\fP and \fBfolder\fP subcommands, see the
subsonic\-cli\-user(1), subsonic\-cli\-folder(1) manual pages.
.fi
.sp
.SH BUGS
.nf
Bugs can be reported to your distribution\(aqs bug tracker or upstream
at \fI\%https://github.com/spl0k/supysonic/issues\fP\&.
.fi
.sp
.SH SEE ALSO
.sp
supysonic\-cli\-user(1), supysonic\-cli\-folder(1)
.SH AUTHOR
Louis-Philippe Véronneau
.\" Generated by docutils manpage writer.
.
```

`supysonic-cli-folder.1`
```
.\" Man page generated from reStructuredText.
.
.TH SUPYSONIC-CLI-FOLDER 1 "2019" "" ""
.SH NAME
supysonic-cli-folder \- Supysonic folder management commands
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.nf
supysonic\-cli folder \fBadd\fP <name> <path>
supysonic\-cli folder \fBdelete\fP <name>
supysonic\-cli folder \fBlist\fP
supysonic\-cli folder \fBscan\fP [\-f] [\-\-background | \-\-foreground] [<name>...]
.fi
.sp
.SH ARGUMENTS
.nf
\fBadd\fP     Add a new folder
\fBdelete\fP  Delete a folder
\fBlist\fP    List all the folders
\fBscan\fP    Scan all or specified folders
.fi
.sp
.SH OPTIONS
.nf
\fB\-f\fP | \fB\-\-force\fP
    Force scan of already known files even if they haven\(aqt changed
.fi
.sp
.nf
\fB\-\-background\fP
    Scan in the background. Requires the daemon to be running
.fi
.sp
.nf
\fB\-\-foreground\fP
    Scan in the foreground, blocking the process while the scan is running
.fi
.sp
.SH EXAMPLES
.sp
To add a new folder to your music library, you can do something like this:
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ supysonic\-cli folder add MyLibrary /home/username/Music
.ft P
.fi
.UNINDENT
.UNINDENT
.sp
Once you\(aqve added a folder, you will need to scan it:
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ supysonic\-cli folder scan MyLibrary
.ft P
.fi
.UNINDENT
.UNINDENT
.SH SEE ALSO
.sp
supysonic\-cli(1), supysonic\-cli\-user(1)
.SH AUTHOR
Louis-Philippe Véronneau
.\" Generated by docutils manpage writer.
.
```
`supysonic-cli-user.1`
```
.\" Man page generated from reStructuredText.
.
.TH SUPYSONIC-CLI-USER 1 "2019" "" ""
.SH NAME
supysonic-cli-user \- Supysonic user management commands
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.nf
supysonic\-cli user \fBadd\fP <user> [\-a] [\-p <password>] [\-e <email>]
supysonic\-cli user \fBdelete\fP <user>
supysonic\-cli user \fBchangepass\fP <user> <password>
supysonic\-cli user \fBlist\fP
supysonic\-cli user \fBsetadmin\fP [\-\-off] <user>
.fi
.sp
.SH ARGUMENTS
.nf
\fBadd\fP         Add a new user
\fBdelete\fP      Delete the user
\fBchangepass\fP  Change the user\(aqs password
\fBlist\fP        List all the users
\fBsetadmin\fP    Give admin rights to the user
.fi
.sp
.SH OPTIONS
.nf
\fB\-a\fP | \fB\-\-admin\fP
    Create the user with admin rights
.fi
.sp
.nf
\fB\-p\fP | \fB\-\-password\fP \fI<password>\fP
    Specify the user\(aqs password
.fi
.sp
.nf
\fB\-e\fP | \fB\-\-email\fP \fI<email>\fP
    Specify the user\(aqs email
.fi
.sp
.nf
\fB\-\-off\fP
    Revoke the admin rights if present
.fi
.sp
.SH EXAMPLES
.sp
To add a new admin user:
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
$ supysonic\-cli user add MyUserName \-a \-p MyAwesomePassword
.ft P
.fi
.UNINDENT
.UNINDENT
.SH SEE ALSO
.sp
supysonic\-cli(1), supysonic\-cli\-folder(1)
.SH AUTHOR
Louis-Philippe Véronneau
.\" Generated by docutils manpage writer.
.
```

You can find the RST files that I'd push here:

https://salsa.debian.org/python-team/applications/supysonic/tree/debian/master/debian/man

**Comment by spl0k** [comment]
We should as well merge them here.

About installing them with `setup.py`, couldn't we use the `data_files` argument? It seems that's what _mutagen_ do.
https://github.com/quodlibet/mutagen/blob/9083c427920c5089812aa12d083b77a45990ff06/setup.py#L240-L243

**Comment by baldurmen** [comment]
Yeah, that's doable but it's a bit icky. My point was more that there is no "proper" way to do it with setup.py https://github.com/pypa/packaging-problems/issues/72

That means it would also add a new dependency (docutils) since we'd need to build the man pages during the setup process.

I'll create a PR for this in the next few days.

**Comment by spl0k** [comment]
🤔 ok. Then we'd better leave the man pages lying around in the repo rather than trying to install them and have it break on some systems.

**Comment by spl0k** [comment]
Merged #157.
